### PR TITLE
Floating point support

### PR DESCRIPTION
**Pull request by jml**
Floating point number support, based on dreid's work.

https://github.com/jml/edn/compare/terml...numbers has a better diff.

### CODE REVIEW

File: edn/tests/test_ast.py [hunk 1]
```diff
@@ -68,6 +70,24 @@ def test_integer(self):
         for edn_str, expected in integers:
             self.assertEqual(edn(edn_str).integer(), expected)
 
+    def test_float(self):
+        floats = (
+            # XXX: How do you do 'exact precision' in Python?
```
Comment (dreid) [inline edn/tests/test_ast.py]: Decimal?

File: edn/_ast.py [hunk 2]
```diff
@@ -148,4 +159,10 @@ def unparse(obj):
     Returns a valid edn string representing 'obj'.
     """
     builder = _Builder()
+    # XXX: Cannot coerce Decimal to term, so we need to wrap it up in a Term,
+    # I think.
+    #
+    # Maybe an Exact() term that's returned only for this.
+    # Maybe change the int part of the rule to yield an Exact term
+    # <int_part>:f 'M' -> Exact(f) | ...
```
Comment (dreid) [inline edn/_ast.py]: This comment is outdated?

Comment (jml) [inline edn/_ast.py]: Fixed.

**Comment by jml** [comment]
If there are no other issues, I'm happy to merge.

**Pull request merged**